src/algorithms/tonal/audio2pitch.h

```diff
@@ -0,0 +1,80 @@
+#ifndef ESSENTIA_AUDIO2PITCH_H
+#define ESSENTIA_AUDIO2PITCH_H
+
+#include "algorithmfactory.h"
+
+namespace essentia {
+namespace standard {
+
+class Audio2Pitch : public Algorithm {
+
+  protected: 
+    Input<std::vector<Real>> _frame;
+    Output<Real> _pitch;
+    Output<Real> _pitchConfidence;
+    Output<Real> _loudness;
+    Output<int> _voiced;
+
+    Algorithm* _pitchAlgorithm;
+    Algorithm* _loudnessAlgorithm;
+    // auxiliary algorithms for FFT-based pitch
+    Algorithm* _windowing;
+    Algorithm* _spectrum;
+
+    Real _sampleRate;
+    int _frameSize;
+    Real _minFrequency;
+    Real _maxFrequency;
+    std::string _pitchAlgorithmName;
+    std::string _loudnessAlgorithmName;
+    Real _tolerance;
+    Real _pitchConfidenceThreshold;
+    Real _loudnessThreshold;
+
+    bool _isSpectral;
+    
+    bool isAboveThresholds(Real pitchConfidence, Real loudness);
+
+  public:
+    Audio2Pitch() {
+      declareInput(_frame, "frame", "the input frame to analyse");
+      declareOutput(_pitch, "pitch", "detected pitch in Hz");
+      declareOutput(_pitchConfidence, "pitchConfidence", "confidence of detected pitch from 0.0 - 1.0");
+      declareOutput(_loudness, "loudness", "detected loudness in dBFS");
+      declareOutput(_voiced, "voiced", "voiced frame categorization, 1 for voiced and 0 for unvoiced frame");
+
+      _isSpectral = true;
+    }
+
+    ~Audio2Pitch() {
+      if (_pitchAlgorithm) delete _pitchAlgorithm;
+      if (_loudnessAlgorithm) delete _loudnessAlgorithm;
+      if (_windowing) delete _windowing;
+      if (_spectrum) delete _spectrum;
+    }
+
+    void declareParameters() {
+      declareParameter("sampleRate", "sample rate of incoming audio frames", "[8000,inf)", 44100);
+      declareParameter("frameSize", "size of input frame in samples", "[1,inf)", 1024);
+      declareParameter("minFrequency", "minimum frequency to detect in Hz", "[10,20000]", 60.f);
+      declareParameter("maxFrequency", "maximum frequency to detect in Hz", "[10,20000]", 2300.f);
+      declareParameter("pitchAlgorithm", "pitch algorithm to use", "{pyin,pyin_fft}", "pyin_fft");
+      declareParameter("loudnessAlgorithm", "loudness algorithm to use", "{loudness,rms}", "rms");
+      declareParameter("weighting", "string to assign a weighting function", "{default,A,B,C,D,Z}", "default");
```

Comment:
By default PitchYinFFT apply a weighting function defined [here](https://github.com/MTG/essentia/blob/8f324eac55440bf02e370b903493a98c6d394660/src/algorithms/tonal/pitchyinfft.cpp#L37) we can think in another name though. Otherwise we can add a comment like `Default PitchYinFFt weighting function` ?

Reply:
@dbogdanov I think this is still pending

Reply:
Maybe we can call it `custom` and use a comment like you suggest: `Custom PitchYinFFt weighting function`

Reply:
Ok, it is done. But this implies to modify pitchyinfft algorithm.